internal/cmds/cmds.go

```diff
@@ -331,21 +331,52 @@ func checkAndSaveSeqNum(ctx log.Logger, seq int, mrseqPath string) (shouldExit b
 	return false, nil
 }
 
+// Copy state of the extension from old version to new version during update (.mrseq files, .status files)
+func CopyStateForUpdate(ctx log.Logger) error {
+	// Copy .mrseq files (Most Recently executed Sequence number) that helps determine whether a sequence number of Run Command has been previously executed or not.
+	err := copyFiles(ctx, ".mrseq", "")
+	if err != nil {
+		return err
+	}
+
+	// Copy .status files of already executed sequence numbers
+	err = copyFiles(ctx, ".status", "status")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
 // Copy *.mrseq (Most Recently executed Sequence number) files from old extension version to new extension version during update.
-func copyMrseqFiles(ctx log.Logger) (error) {	
+func copyFiles(ctx log.Logger, fileExtensionSuffix string, extensionSubdirectory string) error {
 
 	newExtensionVersion := os.Getenv(constants.ExtensionVersionEnvName)
 	oldExtensionVersion := os.Getenv(constants.ExtensionVersionUpdatingFromEnvName)
 
-	ctx.Log("message", fmt.Sprintf("Migrating .mrseq (Most Recently executed Sequence number) files from extension version '%s' to '%s'",oldExtensionVersion, newExtensionVersion ))
-	
+	ctx.Log("message", fmt.Sprintf("Migrating '%s' files from extension version '%s' to '%s'", fileExtensionSuffix, oldExtensionVersion, newExtensionVersion))
+
 	newExtensionDirectory := os.Getenv(constants.ExtensionPathEnvName)
 	oldExtensionDirectory := strings.ReplaceAll(newExtensionDirectory, newExtensionVersion, oldExtensionVersion)
 
+	// Append subdirectory like "status" under extension folder if provided.
+	if extensionSubdirectory != "" {
+		newExtensionDirectory = filepath.Join(newExtensionDirectory, extensionSubdirectory)
+		oldExtensionDirectory = filepath.Join(oldExtensionDirectory, extensionSubdirectory)
+
+		// Create subdirectory like "status" directory if it does not exist
+		_, err := os.Open(newExtensionDirectory)
```

Comment:
does Open succeed on an existing directory?

Reply:
Yes, open works on an existing directory ( I tested it). err would be nil if there is a status directory already and we won't recreate the status directory